projects/owa-data/scripts/conversion/VPT/vpt_mp4_to_mkv.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env -S uv run --script
+# /// script
+# requires-python = ">=3.11"
+# dependencies = [
+#   "mcap-owa-support==0.5.5",
+#   "owa-core==0.5.5",
+#   "tqdm",
+#   "rich",
+# ]
+# [tool.uv]
+# exclude-newer = "2025-08-01T12:00:00Z"
+# ///
+from pathlib import Path
+from tqdm import tqdm
+from rich import print
+from concurrent.futures import ProcessPoolExecutor, as_completed
+
+from owa.core.io.video import VideoReader, VideoWriter
+
+VPT_FOLDER_PATH = Path(
+    "/mnt/raid12/datasets/owa/mcaps/vpt"
+).expanduser()  # NOTE: Change this to your VPT data folder path. We expect paired mp4 and jsonl files for VPT dataset.
+
+
+def process_single_file(mp4_file_path):
+    """Process a single mp4 file and convert it to mkv format."""
+
+    mkv_file_path = mp4_file_path.with_suffix(".mkv")
+
+    # Process VFR to CFR: read with fps sampling, write as CFR
+    target_fps = 20.0
+    with VideoReader(mp4_file_path, force_close=True) as reader:
+        with VideoWriter(mkv_file_path, fps=target_fps, vfr=False) as writer:
+            frame_count = 0
+            for frame in reader.read_frames(fps=target_fps):  # Sample at regular intervals
```

Comment:
Check once specifying fps here yields safe result

Reply:
```
❯ uv run vpt_mp4_to_mkv.py  --test /mnt/raid12/datasets/owa/mcaps/vpt
Running test_target_fps with random file: 
/mnt/raid12/datasets/owa/mcaps/vpt/jumpy-denim-lion-f153ac423f61-20211228-213437.mp4
Processing 
mp4_file_path=PosixPath('/mnt/raid12/datasets/owa/mcaps/vpt/jumpy-denim-lion-f153ac423f61-20211228-213437.mp4')...
opencv frame_count=6001
no target_fps frame_count=6001
target_fps=20.0 frame_count=6001 frame_count/target_fps=300.05
```

Seems ok to me